### PR TITLE
Revised the state machine.

### PR DESCRIPTION
**Pull request by a11r**
Added an IDLE state and replaced FATAL_FAILURE with shutdown. Also modified come phrasing for TRANSIENT_FAILURE for clarity.

### CODE REVIEW

File: doc/connectivity-semantics-and-api.md [hunk 1]
```diff
@@ -34,14 +34,26 @@ retries are done with exponential backoff, channels that fail to connect will
...
-FATAL_FAILURE: There has been a fatal failure and the channel will never
-attempt to establish a connection again.  (e.g., a server presenting an invalid
-TLS certificate)
-
-Channels that enter this state never leave this state.
+because the server is not yet available), the channel may spend increasingly
+large amounts of time in this state.
+
+IDLE: This is the state where the channel is not even trying to create a
+connection because of a lack of new or pending RPCs. New channels MAY be created
```
Comment (ejona86) [inline doc/connectivity-semantics-and-api.md]: Could we have a definition of a "pending RPC"? Is it a synonym of active RPC? What is an active RPC?

File: doc/connectivity-semantics-and-api.md [hunk 2]
```diff
@@ -75,6 +90,15 @@ corresponding reasons. Empty cells denote disallowed transitions.
     <td></td>
     <td></td>
     <td></td>
+    <td>Shutdown triggered by application.</td>
+  </tr>
+  <tr>
+    <th>IDLE</th>
+    <td>Any new RPC activity on the channel</td>
```
Comment (ejona86) [inline doc/connectivity-semantics-and-api.md]: "Any new RPCs on the channel"? If we receive a GOAWAY existing RPCs can continue, but we would still be in IDLE.

File: doc/connectivity-semantics-and-api.md [hunk 3]
```diff
@@ -34,14 +34,26 @@ retries are done with exponential backoff, channels that fail to connect will
...
+because the server is not yet available), the channel may spend increasingly
+large amounts of time in this state.
+
+IDLE: This is the state where the channel is not even trying to create a
+connection because of a lack of new or pending RPCs. New channels MAY be created
+in this state. Any attempt to start an RPC on the channel will push the channel
+out of this state to connecting. When there has been no RPC activity on a channel
+for a specified IDLE_TIMEOUT, i.e., no new or pending (active) RPCs for this
+period, channels that are READY or CONNECTING switch to IDLE. Additionaly,
+channels that receive a GO_AWAY when there are no active or pending RPCs should
```
Comment (ejona86) [inline doc/connectivity-semantics-and-api.md]: Note that HTTP/2 calls it "GOAWAY" (no underscore). I don't know if we care.

File: doc/connectivity-semantics-and-api.md (hunk 1 above)
Comment (a11r) [inline doc/connectivity-semantics-and-api.md]: Synonyms. An active/pending RPC  is one that has been started but has not finshed (returned status). This is from the channel's perspective, so if the channel is wrapped by something and that wrapper buffers RPCs, those buffered RPCs will be active at the wrapper but not at the channel.

File: doc/connectivity-semantics-and-api.md (hunk 2 above)
Comment (a11r) [inline doc/connectivity-semantics-and-api.md]: If there are any pending RPCs, we would not be in idle even if we saw a GOAWAY.

Comment (ejona86) [inline doc/connectivity-semantics-and-api.md]: Then what state would we be in? I'm not sure I understand this document then.

If we get a GOAWAY then we have to reconnect for new RPCs and go into CONNECTING. That reconnect could fail and cause us to be in TRANSIENT_FAILURE. Would we not go through those states?

If we don't go to IDLE, then how do we go to CONNECTING? Go directly there from READY, without going through IDLE/TRANSIENT_FAILURE?

Comment (a11r) [inline doc/connectivity-semantics-and-api.md]: A READY channel that GETS a GOAWAY but has pending RPCs should go to TRANSIENT_FAILURE and then CONNECTING.

Channels with pending RPCs should never enter IDLE as by definition, they are not idle and have some work to do.

**Comment by a11r** [comment]
Updated the spelling of GO_AWAY to GOAWAY and responded to questions. PTAL.

File: doc/connectivity-semantics-and-api.md (hunk 2 above)
Comment (ejona86) [inline doc/connectivity-semantics-and-api.md]: Ah, that makes sense. Thanks!

**Pull request merged**